tapchannel/commitment.go

```diff
@@ -363,20 +363,24 @@ func processAddEntry(htlc *DecodedDescriptor, ourBalance, theirBalance uint64,
 // non-dust satoshi balance. It also checks and returns whether we need a local
 // and/or remote anchor output.
 func SanityCheckAmounts(ourBalance, theirBalance btcutil.Amount,
-	ourAssetBalance, theirAssetBalance uint64, view *DecodedView,
-	chanType channeldb.ChannelType, whoseCommit lntypes.ChannelParty,
-	dustLimit btcutil.Amount) (bool, bool, error) {
+	ourAssetBalance, theirAssetBalance uint64, assetView,
+	nonAssetView *DecodedView, chanType channeldb.ChannelType,
+	whoseCommit lntypes.ChannelParty, dustLimit btcutil.Amount) (bool, bool,
+	error) {
 
 	log.Tracef("Sanity checking amounts, whoseCommit=%v, ourBalance=%d, "+
 		"theirBalance=%d, ourAssetBalance=%d, theirAssetBalance=%d",
 		whoseCommit, ourBalance, theirBalance, ourAssetBalance,
 		theirAssetBalance)
 
 	var (
-		numHTLCs int64
-		feePerKw = view.FeePerKw
+		// When determining whether we need an anchor output, we need to
+		// count the non-asset HTLCs as well.
+		numHTLCs = uint64(len(nonAssetView.OurUpdates) +
```

Comment:
would this lead to stuck force-closes? (covered in the [LitD itest](https://github.com/lightninglabs/lightning-terminal/blob/98c9819b81d03f926f72656d9c0fb33c010e933e/itest/litd_custom_channels_test.go#L3533-L3537)?)

trying to understand if we ever had coverage for this

Reply:
No, the channel will not force close (at least from my observations), but the HTLC will be rejected. This is covered in the new `litd` itest, where we pay sats over an asset channel.

Reply:
Won't this result in us _over counting_? The non asset view will still have an manifest HTLC for a given asset HTLC. I think it's enough to just have this combined sum here, and drop the `numHTLCs` increment in the inner loop.

Reply:
No, the non-asset view and the asset view are fully distinct. And even if we were over counting, we only use `numHTLCs > 0` as a condition for determining whether we need a commitment anchor output.

Reply:
I thought about this again and my initial approach wasn't 100% correct. It is correct that the `assetView` and
`nonAssetView` are non-overlapping. But we need to replicate this behavior:
https://github.com/lightningnetwork/lnd/blob/ab7aae07085ec244f7888e990e8d8cc6ae6dfa83/lnwallet/commitment.go#L706

So we also need to check for dust, which I now added.